### PR TITLE
chore: prefer string slice

### PR DESCRIPTION
**Pull request by septs**
closes #

### CODE REVIEW

File: packages/mask/src/plugins/CryptoartAI/SNSAdaptor/HistoryTab/Row.tsx [hunk 1]
```diff
@@ -84,7 +84,7 @@ export function Row({ event, chainId }: Props) {
             </TableCell>
             <TableCell>
                 <Typography className={classes.content} variant="body2">
-                    {event.createTime.substr(0, event.createTime.length - 3)}
+                    {event.createTime.slice(0, Math.max(0, event.createTime.length - 3))}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/CryptoartAI/SNSAdaptor/HistoryTab/Row.tsx:87]:
```suggestion
                    {event.createTime.slice(0, 0, event.createTime.length - 3)}
```

File: packages/mask/src/plugins/CryptoartAI/SNSAdaptor/OrderRow.tsx [hunk 2]
```diff
@@ -90,7 +90,7 @@ export function OrderRow({ event, chainId }: IRowProps) {
             </TableCell>
             <TableCell>
                 <Typography className={classes.content} variant="body2">
-                    {event.createTime.substr(0, event.createTime.length - 3)}
+                    {event.createTime.slice(0, Math.max(0, event.createTime.length - 3))}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/CryptoartAI/SNSAdaptor/OrderRow.tsx:93]:
```suggestion
                    {event.createTime.slice(0, event.createTime.length - 3)}
```

File: packages/plugins/Flow/src/helpers/formatter.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export function formatAddress(address: string, size = 0) {
     if (!/0x\w{16}/.test(address)) return address
     if (size === 0 || size >= 8) return address
-    return `${address.substr(0, 2 + size)}...${address.substr(-size)}`
+    return `${address.slice(0, Math.max(0, 2 + size))}...${address.slice(-size)}`
```
Comment (guanbinrui) [inline packages/plugins/Flow/src/helpers/formatter.ts:4]:
```suggestion
    return `${address.slice(0, 2 + size)}...${address.slice(-size)}`
```

File: packages/shared/src/utils/formatter.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
 export function formatFingerprint(fingerprint: string, size: number = 0) {
     if (size === 0) return fingerprint
-    return `${fingerprint.substr(0, 2 + size)}...${fingerprint.substr(-size)}`
+    return `${fingerprint.slice(0, Math.max(0, 2 + size))}...${fingerprint.slice(-size)}`
```
Comment (guanbinrui) [inline packages/shared/src/utils/formatter.ts:3]:
```suggestion
    return `${fingerprint.slice(0, 2 + size)}...${fingerprint.slice(-size)}`
```

File: packages/web3-shared/evm/utils/formatter.ts [hunk 5]
```diff
@@ -55,27 +55,27 @@ export function formatEthereumAddress(address: string, size = 0) {
     if (!EthereumAddress.isValid(address)) return address
     const address_ = EthereumAddress.checksumAddress(address)
     if (size === 0 || size >= 20) return address_
-    return `${address_.substr(0, 2 + size)}...${address_.substr(-size)}`
+    return `${address_.slice(0, Math.max(0, 2 + size))}...${address_.slice(-size)}`
```
Comment (guanbinrui) [inline packages/web3-shared/evm/utils/formatter.ts:58]: Does it's better to use `substring` here?

File: packages/shared/src/utils/formatter.ts (hunk 4 above)
Comment (guanbinrui) [inline packages/shared/src/utils/formatter.ts:3]: Does it's better to use `substring` here?

File: packages/mask/src/plugins/CryptoartAI/SNSAdaptor/HistoryTab/Row.tsx (hunk 1 above)
Comment (Jack-Works) [reply packages/mask/src/plugins/CryptoartAI/SNSAdaptor/HistoryTab/Row.tsx:87]: Slice does not accept the third argument.

**Pull request closed without merging**